lib/widgets/read_times_tamp.dart

```diff
@@ -0,0 +1,92 @@
+import 'dart:ui';
+
+import 'package:intl/intl.dart';
```

Comment:
lots of commented code here. 
This class is huge and lots of logic. 

leave comments here explaining this static method, why is needed, how it works. and how the logic works

Reply:
Sorry that was me trying to debug, But I think the error is coming from timesTamp added when a transaction is made, I was not able to fix it but I left it for UI testing. I clean the widget and left only necessary code.